src/app/content/components/ContentExcerpt.tsx

```diff
@@ -30,22 +31,30 @@ const ContentExcerpt = (props: Props) => {
     forwardedRef,
     ...excerptProps
   } = props;
-
+  const services = useServices();
+  const [excerptSourceUrl, setExcerptSourceUrl] = React.useState('');
   const currentBook = assertDefined(useSelector(book), 'book not loaded');
   const sourcePage = typeof source === 'string'
     ? assertDefined(findArchiveTreeNodeById(currentBook.tree, source), 'page not found in book')
     : source;
 
-  const excerptSource = getBookPageUrlAndParams(
-    currentBook,
-    sourcePage
-  );
+  useEffect(() => {
+    const setBookPageUrlAndParams = async() => {
```

Comment:
is it not ok for the effect itself to be async?

Reply:
Making the effect itself to be async triggers eslint warning: `Effect callbacks are synchronous to prevent race conditions. Put the async function inside.` 
There is an [article](https://www.robinwieruch.de/react-hooks-fetch-data) about it.